### PR TITLE
Adding missing fields to API responses

### PR DESCRIPTION
**Pull request by davidjumani**
## Description
Adding missing fields in the following APIs
`osdisplayname` in `listVirtualMachines`
`vpcofferingname` in `listVpcs`
`vpcname` in `listPublicIpAddresses`
`vpcname` in `listPrivateGateways`
`vpcname` in `listVpnGateways`
`templatename, podname` in `listRouters`
`templatename, podname` in `listSystemVms`

Fixes: #4161 

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

### CODE REVIEW

**Comment by davidjumani** [comment]
@blueorangutan package

**Comment by davidjumani** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@shwstppr can you review this?

**Comment by davidjumani** [comment]
@blueorangutan test

File: server/src/main/java/com/cloud/api/query/dao/UserVmJoinDaoImpl.java [hunk 1]
```diff
@@ -201,6 +201,7 @@ public UserVmResponse newUserVmResponse(ResponseView view, String objectName, Us
         userVmResponse.setPublicIp(userVm.getPublicIpAddress());
         userVmResponse.setKeyPairName(userVm.getKeypairName());
         userVmResponse.setOsTypeId(userVm.getGuestOsUuid());
+        userVmResponse.setOsDisplayName(ApiDBUtils.findGuestOSById(userVm.getGuestOsId()).getDisplayName());
```
Comment (rhtyd) [inline server/src/main/java/com/cloud/api/query/dao/UserVmJoinDaoImpl.java:204]: Add defensive check for potential NPE fix

**Comment by davidjumani** [comment]
@blueorangutan package

**Comment by davidjumani** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@blueorangutan test

**Pull request merged**